src/clib/make_consistent.cpp

```diff
@@ -0,0 +1,1069 @@
+            Sg[i] = 0.;
+            Fet[i] = 0.;
+            Feg[i] = 0.;
+            for (iSN = 0; iSN < nSN; iSN++) {
+              iSN0 = SN_i[iSN];
+
+              Ct[i] = Ct[i] + my_rates->SN0_XC[iSN0] * SN_metal(i, iSN);
+              Ot[i] = Ot[i] + my_rates->SN0_XO[iSN0] * SN_metal(i, iSN);
+              Mgt[i] = Mgt[i] + my_rates->SN0_XMg[iSN0] * SN_metal(i, iSN);
+              Alt[i] = Alt[i] + my_rates->SN0_XAl[iSN0] * SN_metal(i, iSN);
+              Sit[i] = Sit[i] + my_rates->SN0_XSi[iSN0] * SN_metal(i, iSN);
+              St[i] = St[i] + my_rates->SN0_XS[iSN0] * SN_metal(i, iSN);
+              Fet[i] = Fet[i] + my_rates->SN0_XFe[iSN0] * SN_metal(i, iSN);
+
+              Cg[i] = Cg[i] + my_rates->SN0_fC[iSN0] * SN_metal(i, iSN);
+              Og[i] = Og[i] + my_rates->SN0_fO[iSN0] * SN_metal(i, iSN);
+              Mgg[i] = Mgg[i] + my_rates->SN0_fMg[iSN0] * SN_metal(i, iSN);
+              Alg[i] = Alg[i] + my_rates->SN0_fAl[iSN0] * SN_metal(i, iSN);
+              Sig[i] = Sig[i] + my_rates->SN0_fSi[iSN0] * SN_metal(i, iSN);
+              Sg[i] = Sg[i] + my_rates->SN0_fS[iSN0] * SN_metal(i, iSN);
+              Feg[i] = Feg[i] + my_rates->SN0_fFe[iSN0] * SN_metal(i, iSN);
+            }
+          }
+        }
+
+        for (i = my_fields->grid_start[0]; i <= my_fields->grid_end[0]; i++) {
+          Cd[i] = Ct[i] - Cg[i];
+          Od[i] = Ot[i] - Og[i];
+          Mgd[i] = Mgt[i] - Mgg[i];
+          Ald[i] = Alt[i] - Alg[i];
+          Sid[i] = Sit[i] - Sig[i];
+          Sd[i] = St[i] - Sg[i];
+          Fed[i] = Fet[i] - Feg[i];
+        }
+
+        for (i = my_fields->grid_start[0]; i <= my_fields->grid_end[0]; i++) {
+          // if (itmask_metal(i)) then
+          OH(i, j, k) = std::fabs(OH(i, j, k));
+          H2O(i, j, k) = std::fabs(H2O(i, j, k));
+          CH(i, j, k) = std::fabs(CH(i, j, k));
+          CH2(i, j, k) = std::fabs(CH2(i, j, k));
+          OHII(i, j, k) = std::fabs(OHII(i, j, k));
+          H2OII(i, j, k) = std::fabs(H2OII(i, j, k));
+          H3OII(i, j, k) = std::fabs(H3OII(i, j, k));
+          totalH[i] = totalH[i] + OH(i, j, k) / 17. + H2O(i, j, k) / 18. * 2. +
+                      CH(i, j, k) / 13. + CH2(i, j, k) / 14. * 2. +
+                      OHII(i, j, k) / 17. + H2OII(i, j, k) / 18. * 2. +
+                      H3OII(i, j, k) / 19. * 3.;
+          // endif
+        }
+      }
+
+      // Correct densities by keeping fractions the same
+
+      for (i = my_fields->grid_start[0]; i <= my_fields->grid_end[0]; i++) {
+        correctH =
+            (gr_float)(my_chemistry->HydrogenFractionByMass * metalfree[i] /
+                       totalH[i]);  // TODO: which one is correct?
+        // correctH = (gr_float)(my_chemistry->HydrogenFractionByMass*(1. -
+        // my_chemistry->DeuteriumToHydrogenRatio)*metalfree[i]/totalH[i]
+        //              );
+        // // ! GC202005
+        // //- !       correctH = real(fh*metalfree(i)/totalH(i), RKIND)
```

Comment:
What's going on here? Did you manually add these comments yourself?

This logic doesn't seem to be present in the original version of the function...